lib/src/components/NxForm/stateful/__tests__/NxStatefulForm.test.tsx

```diff
@@ -350,77 +354,28 @@ describe('NxStatefulForm', function() {
 
     await user.click(screen.getByRole('button', { name: 'Submit' }));
```

Comment:
Could you please explain why we have `await user.click on Submit button` here, is it to test that clicking on Submit button will not set `showValidationErrors` to true when there is no validationErrors? 
Seems like the line# 357 will still pass without this.

Reply:
> Seems like the line# 357 will still pass without this.

Yes, but we want to make sure it passes _even with_ this.  As the two tests above this one establish, `showValidationErrors` always starts false and only gets set to true when the submit button is clicked and the `validationErrors` prop has certain types of values.  In this test here, I basically am testing a wide variety of transitions between different kinds of `validationErrors` values. For each transition where `showValidationErrors` was already `false`, I have to click the submit button to find out whether the transition did anything, because even if that transition would change `showValidationErrors`, it won't do so until the button is clicked.

Reply:
Thanks for the explanation.